### DISCUSSION TITLE
TreeView with different classes

### DISCUSSION THREAD

**Discussion by Agredo** [AvaloniaUI/Avalonia#12155] [Q&A]
I want to to a TreeViewer for following List<A> listA

class A
{
prop string Name ...;
prop ICollection<B> ListB ..;
//Other Props
}

class B
{
prop string Name ...;
prop ICollection<C> ListC ..;
//Other Props
}

class C
{
prop string Name ...;
//Other Props
}

Is there a simple way to use the NameProperies of all these classes in a TreeView in MVVM?

This works for the A but for B ist already only showing the class Name

`
 <TreeView ItemsSource="{Binding ListA}">
                <TreeView.ItemTemplate>
                    <TreeDataTemplate x:DataType="models:A" ItemsSource="{Binding ListB}">
                       <TextBlock Text="{Binding Name}"/>
                    </TreeDataTemplate>
                </TreeView.ItemTemplate>
</TreeView>
`

**Comment by timunie** [comment]
Hm... you can probably extract a base class or interface for it. So the DataTemplate can be done using this interface. Otherwise maybe you need to implement your own TreeDataTemplate.

**Reply by Agredo** [reply]
Thank you I would like it more to avoid this but okay when this is the most simple solution its okay. 🙂
